### PR TITLE
Fix for FloatToInt

### PR DESCRIPTION
**Pull request by DanielGibson**
I screwed up handling negative numbers. Oops.
Luckily GCC told me about it (only used clang before).

Also improved warning suppression in cmake a bit.

### CODE REVIEW

**Comment by rohit-n** [comment]
If the plan is to remove suppressions one day, should I get started on removing unused (and possibly set) variables?

**Comment by icculus** [comment]
> If the plan is to remove suppressions one day, should I get started on removing unused (and possibly set) variables?

Well, it's a thankless job, but if you want to do that, that would be awesome!

File: Sources/CMakeLists.txt [hunk 1]
```diff
@@ -112,9 +112,20 @@ if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID STREQUAL "Clang")
 	# TODO fix these warnings
 	add_compile_options(-Wno-sign-compare)
 	add_compile_options(-Wno-switch)
-	add_compile_options(-Wno-format-security)
-	add_definitions(-Wno-logical-op-parentheses)
-	MESSAGE(WARNING, "re-enable -Wlogical-op-parentheses some day!")
+	add_compile_options(-Wno-unknown-pragmas)
+	add_compile_options(-Wno-unused-variable)
+	add_compile_options(-Wno-unused-value)
+	add_compile_options(-Wno-reorder)
```
Comment (icculus) [inline Sources/CMakeLists.txt:118]: I think someone cleaned this reorder warning issues out the other day, so we might not need this one now. I haven't tested personally to see, though.

Comment (rohit-n) [inline Sources/CMakeLists.txt:118]: That was me :)

**Pull request merged**